lib/claim_letters/claim_letter_downloader.rb

```diff
@@ -68,12 +71,13 @@ def format_letter_data(docs)
       letters.sort_by { |d| d[:received_at] }.reverse
     end
 
-    def verify_letter_in_folder(document_id)
+    def get_letter_details(document_id)
       letters = get_letters
+      letters.select { |d| d[:document_id] == document_id }
```

Comment:
Will we ever want to return an array here? This may be a rusty ruby reaction, but it seems to me like we could use a `find` instead and more clearly specify that we're returning a single letter. This would somewhat simplify the code above as well (I'll comment up there to be consistent).